### PR TITLE
Allow data explorer to use connect

### PR DESCRIPTION
**Pull request by anthonydresser**
Adds the necessary changes to enable the data explorer to utilize the connect method which gets the connection dialog and firewall rule perks.

### CODE REVIEW

File: src/sql/platform/connection/common/connectionManagement.ts [hunk 1]
```diff
@@ -57,7 +57,7 @@ export interface IConnectionResult {
 export interface IConnectionCallbacks {
 	onConnectStart(): void;
 	onConnectReject(error?: string): void;
-	onConnectSuccess(params?: INewConnectionParams): void;
+	onConnectSuccess(params: INewConnectionParams, profile: IConnectionProfile): void;
```
Comment (anthonydresser) [inline src/sql/platform/connection/common/connectionManagement.ts]: The connection dialog can change the profile which means the connection management service and the caller are referring to different profiles. This allows the the caller to get the updated profile.

File: src/sql/platform/connection/common/connectionManagement.ts [hunk 2]
```diff
@@ -80,7 +80,7 @@ export interface IConnectionManagementService {
 	/**
 	 * Opens the connection dialog to create new connection
 	 */
-	showConnectionDialog(params?: INewConnectionParams, model?: IConnectionProfile, connectionResult?: IConnectionResult): Promise<void>;
+	showConnectionDialog(params?: INewConnectionParams, options?: IConnectionCompletionOptions, model?: IConnectionProfile, connectionResult?: IConnectionResult): Promise<void>;
```
Comment (anthonydresser) [inline src/sql/platform/connection/common/connectionManagement.ts]: The connection management service was ignoring the connection options when it hits the dialog which means it looses the settings defined by the caller.

File: src/sql/platform/connection/common/connectionManagementService.ts [hunk 3]
```diff
@@ -471,7 +471,11 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 						connection.saveProfile = false;
 						this.doActionsAfterConnectionComplete(uri, options);
 					}
-					resolve(connectionResult);
+					if (connection.savePassword) {
```
Comment (anthonydresser) [inline src/sql/platform/connection/common/connectionManagementService.ts]: save password was only being respected if we saved the profile, which doesn't work with the data explorer.

File: src/sql/platform/connection/common/connectionProfile.ts [hunk 4]
```diff
@@ -41,6 +43,12 @@ export class ConnectionProfile extends ProviderConnectionInfo implements interfa
 			this.saveProfile = model.saveProfile;
 			this._id = model.id;
 			this.azureTenantId = model.azureTenantId;
+			const options = this.capabilitiesService.getCapabilities(model.providerName).connection.connectionOptions;
+			let appNameOption = options.find(option => option.specialValueType === ConnectionOptionSpecialType.appName);
```
Comment (anthonydresser) [inline src/sql/platform/connection/common/connectionProfile.ts]: The appname is added by the dialog, which means in cases where we don't hit the dialog the profile won't match for matching cases.

**Comment by kburtram** [comment]
It looks like the CI builds are failing.  We should investigate prior to merging.  Thanks!

**Pull request merged**